test_databaseconnect.py

```diff
@@ -0,0 +1,151 @@
+        try:
+            test_db = mysql.connect(
+                host=config.host,
+                user=config.user,
+                passwd=config.password
+            )
+
+            cursor = test_db.cursor()
+            cursor.execute("CREATE DATABASE {}".format(config.database))
+            print("test database created")
+
+        except Exception:
+            print ("Failed to create test database")
+            # rolling back to orignal db
+            config.database = os.getenv("mapbotdatabase")
+            pytest.exit("Exiting pytest")
+
+        yield test_db
+
+        # Will be executed after the last test
+        try:
+            mycursor = test_db.cursor()
+            mycursor.execute("DROP DATABASE {}".format(config.database))
+            mycursor.close()
+            print("test database deleted.")
+
+        except Exception:
+            print ("Failed to delete test databse.")
+
+        config.database = os.getenv("mapbotdatabase")
+
+
+    def test_connection_to_database(self):
+        with pytest.raises(Exception) as e:
+            raise (e)
+        result = databaseconnect.connection_to_database()
+        assert type(result) == mysql.connection_cext.CMySQLConnection
+
+
+    def test_setup_database(self):
+        db = databaseconnect.setup_database()
+        cursor = db.cursor()
+        cursor.execute("SHOW TABLES")
+        tables = cursor.fetchall()
+        expected_tables = [("chat_table"), ("statement_table"), ("question_table"), ("directions_table"),]
+        assert tables.sort() == expected_tables.sort()           
+
+
+    def test_add_to_database_of_chat_table(self):
+        db = databaseconnect.add_to_database("C", "subject", "root", "verb", "H")
+        cursor = db.cursor()
+        cursor.execute("select * from chat_table where root_word='root' and verb='verb' and sentence='H'")        
+        assert type(cursor.fetchone()) is tuple
+
+
+    def test_add_to_database_of_question_table(self):
+        db = databaseconnect.add_to_database("Q", "subject", "root", "verb", "H")
+        cursor = db.cursor()
+        cursor.execute("select * from question_table where subject='subject' and root_word='root' and verb='verb' and sentence='H'")        
+        assert type(cursor.fetchone()) is tuple
+
+    
+    def test_add_to_database_of_statement_table(self):
+        db = databaseconnect.add_to_database("O", "subject", "root", "verb", "H")
+        cursor = db.cursor()
+        cursor.execute("select * from statement_table where subject='subject' and root_word='root' and verb='verb' and sentence='H'")        
+        assert type(cursor.fetchone()) is tuple
+
+
+    def test_get_chat_response(self):
+        response = databaseconnect.get_chat_response()
+        assert type(response) is str
+
+
+    def test_get_question_response_without_subject(self):
+        response = databaseconnect.get_question_response("[]", "root", "verb")
+        assert type(response) is tuple
```

Comment:
This type check still makes sense since the types and functionalities are defined by us. A few other assertions can be added on top of this one.

Reply:
other assertions like?